numpydoc/numpydoc.py

```diff
@@ -203,12 +203,29 @@ def mangle_signature(app, what, name, obj, options, sig, retann):
     if not hasattr(obj, '__doc__'):
         return
     doc = get_doc_object(obj, config={'show_class_members': False})
-    sig = doc['Signature'] or getattr(obj, '__text_signature__', None)
+    sig = (doc['Signature']
+           or _clean_text_signature(getattr(obj, '__text_signature__', None)))
     if sig:
         sig = re.sub("^[^(]*", "", sig)
         return sig, ''
 
 
+def _clean_text_signature(sig):
+    if sig is None:
+        return None
+    start_pattern = re.compile(r"^[^(]*\(")
+    start, end = start_pattern.search(sig).span()
+    start_sig = sig[start:end]
+    sig = sig[end:-1]
+    sig = re.sub(r'^\$(self|module|type)(,\s|$)','' , sig, count=1)
+    def slashfunc(matchobj):
+        if matchobj.group(0)[0] == '/':
+            return '*'
+        return ', *'
+    sig = re.sub(r'(^|,\s)/,\s\*', slashfunc, sig, count=1)
```

Comment:
Here are two options that would suffice without a callback:
```suggestion
    sig = re.sub(r'(^|(?<=,\s))/,\s\*', '*', sig, count=1)
```
or
```suggestion
    sig = re.sub(r'(^|,\s)/(?=,\s\*)', '', sig, count=1)
```
These use lookahead or lookbehind to require but not match some parts.

Reply:
Ahh. Genius.